### ISSUE TITLE
Strange behavior after deleting all (partially) watched videos in playlists

### ISSUE THREAD

**Issue by vkay94** [TeamNewPipe/NewPipe#5169]
### Checklist
- [x] I am using the latest version - 0.20.6 
- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one bug. I will open one issue for every bug report I want to file.

### Steps to reproduce the bug
1. Create a playlist with some (partially) watched videos and not-watched ones.
2. Open the options menu on the top and choose "Yes and partially watched videos" (third option)

3a. Start to play one video.
3b. Open the options menu and choose "Yes" (first option)
 
### Actual behaviour

After 3a: The video is removed from the list immediately and all items are updating very often.
After 3b: The old items are appearing again (even if the optiion says "it can't be undone xD)

### Expected behavior

3a: The video should not disappear/deleted from list.
3b: The videos shouldn't re-appear again (in this case there shouldn't be any difference after third option action)

### Screenshots/Screen recordings
<img src="https://user-images.githubusercontent.com/6626210/102079391-e6582c80-3e0c-11eb-8e53-d3d1a17051d3.gif" width="350">

### Device info

 - Android version/Custom ROM version: Android 10 (LineageOS 17.1)
 - Device model: S4

#### PS:
I've already looked into it a little bit and one reason could be that the remove action doesn't delete items from the database and 
which is the reason for 3b (after deleting only references instead of the actual items from the database, another loading from the table will load the old list).

You can assign me to this issue, @opusforlife2 , but I can already tell that there might be some structure changes.

**Comment by mhmdanas** [comment]
I tried reproducing this (and succeeded), but I actually also found that leaving the app and then re-opening it actually removes the "restored videos." Also, interestingly, videos I watched and finished before are removed but not "restored."

**Comment by vkay94** [comment]
Yeah, everything related to this action is pretty weird. I faced it during trying to change how local lists are updated (moving to `ListAdapter` using `Groupie`: #4739). Pretty much every fragment is working fine except this specific action :')

**Comment by vkay94** [comment]
Actually, it was easier than expected xD
